angular-chart.js

```diff
@@ -1,243 +1,302 @@
 (function () {
   'use strict';
 
-  Chart.defaults.global.responsive = true;
-  Chart.defaults.global.multiTooltipTemplate = '<%if (datasetLabel){%><%=datasetLabel%>: <%}%><%= value %>';
-
-  Chart.defaults.global.colours = [
-    '#97BBCD', // blue
-    '#DCDCDC', // light grey
-    '#F7464A', // red
-    '#46BFBD', // green
-    '#FDB45C', // yellow
-    '#949FB1', // grey
-    '#4D5360'  // dark grey
-  ];
-
   angular.module('chart.js', [])
-    .directive('chartBase', function () { return chart(); })
-    .directive('chartLine', function () { return chart('Line'); })
-    .directive('chartBar', function () { return chart('Bar'); })
-    .directive('chartRadar', function () { return chart('Radar'); })
-    .directive('chartDoughnut', function () { return chart('Doughnut'); })
-    .directive('chartPie', function () { return chart('Pie'); })
-    .directive('chartPolarArea', function () { return chart('PolarArea'); });
-
-  function chart (type) {
-    return {
-      restrict: 'CA',
-      scope: {
-        data: '=',
-        labels: '=',
-        options: '=',
-        series: '=',
-        colours: '=?',
-        getColour: '=?',
-        chartType: '=',
-        legend: '@',
-        click: '=',
-        hover: '='
-      },
-      link: function (scope, elem/*, attrs */) {
-        var chart, container = document.createElement('div');
-        container.className = 'chart-container';
-        elem.replaceWith(container);
-        container.appendChild(elem[0]);
-
-        if (typeof window.G_vmlCanvasManager === 'object' && window.G_vmlCanvasManager !== null) {
-          if (typeof window.G_vmlCanvasManager.initElement === 'function') {
-            window.G_vmlCanvasManager.initElement(elem[0]);
-          }
-        }
+    .provider('ChartJs', ChartJsProvider)
+    .factory('ChartJsFactory', ['ChartJs', ChartJsFactory])
+    .directive('chartBase',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory(); }])
```

Comment:
well that's only an annotation for the minification and can be automated by https://www.npmjs.com/package/gulp-ng-annotate